### DISCUSSION TITLE
Awesome Sanity Studio v3

### DISCUSSION THREAD

**Discussion by kmelve** [sanity-io/sanity#3331] [Show and tell]
Have you made something cool/fun/interesting/useful with Sanity Studio v3? Tell us about it!

**Comment by SimeonGriggs** [comment]
Sanity Studio embedded in a Remix app – with a Sanity UI designed, WordPress-inspired Admin Bar on the front end 🙌 

https://www.loom.com/share/8d3a567d49a44b22ae688c214ca41a4c

**Reply by matt416** [reply]
Very cool

**Reply by mittnavnermike** [reply]
So nice 😍  Would love to see the code for this

**Comment by saiichihashimoto** [comment]
Made [sanity-typed-schema-builder](https://www.npmjs.com/package/sanity-typed-schema-builder)! Creating schemas will get you typescript types of your documents, mocks for testing, and a zod parser for parsing and transforming data, notably getting `Date` out of the datetime types.

**Reply by tbrasington** [reply]
This is really great. I am struggling with the block component though. Getting this type error `The inferred type of 'PageSchema' cannot be named without a reference to 'sanity-typed-schema-builder/node_modules/type-fest'. This is likely not portable. A type annotation is necessary.ts`

**Reply by saiichihashimoto** [reply]
Hmmm, weird. ‘type-fest’ is one of the deps I’m using but I’m not sure why it wouldn’t be there for you. Just to check, do you have the block inside of an array?

**Reply by tbrasington** [reply]
Thank you for getting back to me. Yep, this is how I had it structured

```typescript

export const PageSchema = s.document({
 name: "page",
 title: "Page",
   fields: [
    {
       name: "title",
      title: "Title",
       type: s.string()
     },

     {
       name: "body",
       title: "Body",
       type: s.array({
         of: [
           s.block({
             styles: [
               { title: "Normal", value: "normal" },
               { title: "H1", value: "h1" },
               { title: "H2", value: "h2" },
               { title: "H3", value: "h3" },
               { title: "H4", value: "h4" },
               { title: "Quote", value: "blockquote" }
             ],
             lists: [{ title: "Bullet", value: "bullet" }],
             
             marks: {
               decorators: [
                 { title: "Strong", value: "strong" },
                 { title: "Emphasis", value: "em" }
               ],
               annotations: [
                 {
                   title: "URL",
                   name: "link",
                   type: s.object({
                     fields: [
                       {
                         title: "URL",
                         name: "href",
                         type: s.url({})
                       }
                     ]
                   })
                 }
               ]
             }
           }),
           s.image({ options: { hotspot: true } })
         ]
       })
     }
   ],

   preview: {
     select: {
       title: "title",
       media: "opengraphImage"
     }
   }
 });
```

One thing that I can think of is that I am using this within Turborepo so there maybe some weird package dependency issues going on.

**Reply by saiichihashimoto** [reply]
weird... i don't even really understand what that error really means. `type-fest` is in `dependencies`, so I'm not sure what could be missing? For now, I'd install `type-fest` yourself which will probably fix it but... super weird.

**Reply by mattddean** [reply]
@saiichihashimoto I'm a big fan of the direction of this plugin, but I'm having trouble getting the Studio to recognize a schema it creates. What version of the `sanity` npm package are you on? I'm on `"sanity": "3.0.0-dev-preview.22"` and `"sanity-typed-schema-builder": "1.15.11"`

**Reply by mattddean** [reply]
I tried downgrading to `3.0.0-dev-preview.0` and it's recognizing my schema now. Since it seems like Sanity Studio is rapidly changing while in dev preview, maybe you'd consider making a note of the latest working version of the dev preview in your package's readme?

**Reply by saiichihashimoto** [reply]
I haven't been keeping it up with the new versions, they're iterating so fast! I think I defined specific versions of the dev preview, maybe not.

**Comment by ghost** [comment]
I just completed my new portfolio site with v3 of the Studio. Here's two articles I wrote about Sanity too. Great work to the community!
[I rebuilt my portfolio site (again) – here's what I learned](https://maverickcoryell.com/insights/new-portfolio-site) 
[Why content management was painful before Sanity](https://maverickcoryell.com/insights/sanity-cms)

**Comment by stipsan** [comment]
Here's [Themer](https://themer.creativecody.dev/):
<table><tr>
<td><img width="1392" alt="Screenshot 2022-07-18 at 14 38 13" src="https://user-images.githubusercontent.com/81981/179519552-a75b5fac-755d-4fc9-915b-1cea29f3e35c.png"></td>
<td><img width="1392" alt="Screenshot 2022-07-18 at 14 33 07" src="https://user-images.githubusercontent.com/81981/179519557-d0b4756e-45f2-4783-a4a3-bc0f6cbbd623.png"></td>
<td><img width="1392" alt="Screenshot 2022-07-18 at 14 33 00" src="https://user-images.githubusercontent.com/81981/179519559-69df7b64-b8a2-4eba-b8cd-37946b07c7db.png"></td>
</tr></table>

> [Repo](https://github.com/stipsan/themer)
